### PR TITLE
Test e2e/edit profile

### PR DESCRIPTION
**Pull request by MHerholdt94**
## Title

#1008 

## Describe your changes: 

Created an Cypress E2E test to check that profile details are properly edited when profile edit form is submitted

## Checklist before requesting a review

* [x] I created a new branch before starting with this ticket?
* [x] I commented the issue after finish in the Frontend Team Board?
* [x] I moved the card to Quality Assurance after finish?

## Did you run into any issues?

- No, issues encountered.

## Screenshots:

- No screenshots to include.

## Feedback

- Yes, I want feedback.

### CODE REVIEW

File: cypress/e2e/edit-profile.cy.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get('#email').type('thisuser@stud.noroff.no', { force: true });
+    cy.get('#password').type('qwe123***', { force: true });
+    cy.get('button[data-auth="login"]').click({ force: true });
+
+    // Navigate to profile edit modal
+    cy.contains('Profile').click();
+    cy.contains('EDIT PROFILE').click();
+
+    // Fill out the edit profile form
+    cy.wait(200);
```
Comment (kyrregjerstad) [inline cypress/e2e/edit-profile.cy.js:24]: using .wait is in general not recommended, since we don't know how long the server will take to respond. A better pattern is to assert something that only exists on the page you're expecting. When doing so, cypress will look for the specific element until it timeouts, which is usually a better approach.

File: cypress/e2e/edit-profile.cy.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get('#firstNameStudent').clear().type('This', { delay: 100 });
```
Comment (kyrregjerstad) [inline cypress/e2e/edit-profile.cy.js:25]: This should be split into two lines to solve the lint error. Also, why is the delay needed?


```suggestion
    cy.get('#firstNameStudent').clear()
    cy.get('#firstNameStudent').type('This', { delay: 100 });
```

File: cypress/e2e/edit-profile.cy.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get('#lastNameStudent').clear().type('User', { delay: 100 });
+    cy.get('#studentSkills').clear().type('E2E-testing, Cypress, JavaScript');
+    cy.get('#studentDescription')
+      .clear()
+      .type(`Hello! This "About Me" section is written with a Cypress E2E test`);
+
+    // Submit the form
+    cy.get('.d-grid > .fw-semibold').click();
```
Comment (kyrregjerstad) [inline cypress/e2e/edit-profile.cy.js:33]: this aligns more with what a user sees:
```suggestion
    cy.get('button').contains('SAVE').click();
```

File: cypress/e2e/edit-profile.cy.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    // Wait for the PUT request to complete and check the status code
+    cy.wait('@editProfile').its('response.statusCode').should('eq', 200);
+
+    // Verify updated profile details
+    cy.visit('http://127.0.0.1:4173/pages/user/'); // Please replace the first part with your local vite url
```
Comment (kyrregjerstad) [inline cypress/e2e/edit-profile.cy.js:39]:
```suggestion
    cy.reload();
```

File: cypress/e2e/edit-profile.cy.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get('#profileName').should('contain', 'This User');
+    cy.get('#skillsList > :nth-child(1)').should('contain', 'E2E-testing');
```
Comment (kyrregjerstad) [inline cypress/e2e/edit-profile.cy.js:41]: A bit more robust selector, same goes for the rest of these:
```suggestion
    cy.get('#skillsList').contains('E2E-testing');
```

File: cypress/e2e/edit-profile.cy.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get('#studentSkills').clear().type('E2E-testing, Cypress, JavaScript');
+    cy.get('#studentDescription')
+      .clear()
+      .type(`Hello! This "About Me" section is written with a Cypress E2E test`);
+
+    // Submit the form
+    cy.get('.d-grid > .fw-semibold').click();
+
+    // Wait for the PUT request to complete and check the status code
+    cy.wait('@editProfile').its('response.statusCode').should('eq', 200);
```
Comment (kyrregjerstad) [inline cypress/e2e/edit-profile.cy.js:36]: Is this assertion needed? Checking API status codes seems a bit out of scope for an E2E test, since it's supposed to mimic "user behaviour", and most users don't check their network tab ;)
If the items on the page are there after the refresh, we can assert that the updated worked, but from a more "user-centric" perspective.

**Review by kyrregjerstad — CHANGES_REQUESTED** [review]
Great job on the test setup – it's clear and efficient! 👏
However, I've noticed a potential issue for future runs. Since the test changes the user profile details and then checks these changes, these updates will be present in subsequent runs. This makes it difficult to know if the test is actually making new updates or just finding the already modified details. To ensure clarity, I suggest adding an extra step at the end to either revert the changes or alter the details in a different way. This will help confirm that each test run effectively updates the profile.

Additionally, I've added some code comments with suggestions on how to clean up the code and incorporate some best practices. Let me know if you have any questions!

File: cypress/e2e/edit-profile.cy.js (hunk 6 above)
Comment (MHerholdt94) [reply cypress/e2e/edit-profile.cy.js:37]: Good point! I will remove it

File: cypress/e2e/edit-profile.cy.js (hunk 1 above)
Comment (MHerholdt94) [reply cypress/e2e/edit-profile.cy.js:24]: Could i use `.should("be.visible")` for this? Without .wait the test wont properly type in the firstname field

Comment (kyrregjerstad) [reply cypress/e2e/edit-profile.cy.js:24]: ah - i remember i had this issue with the bootstrap modal as well before! Didn't find a better solution for it in the end, so let's stick with the wait() unless you got any better ideas :)

Comment (MHerholdt94) [reply cypress/e2e/edit-profile.cy.js:24]: I suspect it is because of the very short animation for when the modal pops up. I'll stick to`.wait()` at least for now :p

File: cypress/e2e/edit-profile.cy.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
+describe('Edits profile details', () => {
+  it('should check that profile details are successfully edited', () => {
+    // Intercept the PUT request and login API call
+    cy.intercept('PUT', 'https://cors.noroff.dev/https://agency-api.noroff.dev/users/*').as(
```
Comment (kyrregjerstad) [inline cypress/e2e/edit-profile.cy.js:4]: Good job in cleaning it up! 🚀 
Now that we got the new assertions, it looks like we no longer need the 'intercept''s anymore

File: cypress/e2e/edit-profile.cy.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+    // Navigate to profile edit modal again
+    cy.contains('Profile').click();
+    cy.contains('EDIT PROFILE').click();
+
+    // Fill out the edit profile form again to assure changes happen
+    cy.wait(200);
+    cy.get('#editProfileModal').should('be.visible');
+    cy.get('#firstNameStudent').clear();
+    cy.get('#firstNameStudent').type('Test', { delay: 100 });
```
Comment (kyrregjerstad) [inline cypress/e2e/edit-profile.cy.js:52]: These need to be set back to the original cypress test values, since the `login.cy.js` test depends on those values. firstname: cypress, lastname: test

**Comment by MHerholdt94** [comment]
@kyrregjerstad `intercept`'s are removed and the `firstname` and `lastname` are changed to be `cypress test`

**Comment by kyrregjerstad** [comment]
> @kyrregjerstad `intercept`'s are removed and the `firstname` and `lastname` are changed to be `cypress test`

Nice! But the tests are failing now when you run them a 2nd time, since the first part of the test is looking for the text `This User`

```
AssertionError: Timed out retrying after 4000ms: expected '<h1#profileName.text-center.mt-3.fw-bold>' to contain 'This User'
```

**Comment by kyrregjerstad** [comment]
@MHerholdt94 I played around with the tests a bit, asserting that the `cy.contains('Profile Updated').should('be.visible');` seems to do the trick. This is only visible when the server returns a 200 status, so we know that the server has received the update before refreshing the page. 
I'll merge the branch now so we can check it of as done before the end of the sprint :)

**Review by kyrregjerstad — APPROVED** [review]
